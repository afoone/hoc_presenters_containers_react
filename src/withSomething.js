const withSomething = Comp => props => {
  const {
    object = {
      a: 1,
    },
    color = 'red',
    ...rest
  } = props
  return (
    <div style={{ color }} object={object}>
      <Comp {...rest} />
    </div>
  )
}

export default withSomething
