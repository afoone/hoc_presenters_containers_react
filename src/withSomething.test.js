import { render } from '@testing-library/react'
import withSomething from './withSomething'

describe('withSomething', () => {
  const MockWithColorComponent = withSomething(props => (
    <div>Test Color {props.style}</div>
  ))

  it('should return a function', () => {
    const { container } = render(<MockWithColorComponent />)
    expect(container).toMatchSnapshot()
  })
})
