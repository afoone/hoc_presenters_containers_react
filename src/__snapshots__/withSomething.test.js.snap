// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withSomething should return a function 1`] = `
<div>
  <div
    object="[object Object]"
    style="color: red;"
  >
    <div>
      Test Color 
    </div>
  </div>
</div>
`;
